### PR TITLE
Sample using @reduxjs/toolkit

### PR DESCRIPTION
**Pull request by jespino**
this is a POC and needs some extra work before is merged.

### CODE REVIEW

File: webapp/src/store/currentWorkspace.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const currentWorkspaceSlice = createSlice({
+    name: 'currentWorkspace',
+    initialState: {value: null} as {value: IWorkspace|null},
+    reducers: {
+        setWorkspace: (state, action: PayloadAction<IWorkspace>) => {
+            state.value = action.payload
+        },
+    },
+    extraReducers: (builder) => {
+        builder.addCase(fetchCurrentWorkspace.fulfilled, (state, action) => {
```
Comment (mgdelacroix) [inline webapp/src/store/currentWorkspace.ts:25]: Why not typing the action here? (same in the other `extraReducers`)

File: webapp/src/store/language.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import {RootState} from './index'
+
+export const fetchLanguage = createAsyncThunk(
+    'language/fetch',
+    async () => getCurrentLanguage(),
+)
+
+export const storeLanguage = createAsyncThunk(
+    'language/store',
+    async (lang: string) => {
```
Comment (mgdelacroix) [inline webapp/src/store/language.ts:17]: This function doesn't seem to need `async`:

```suggestion
    (lang: string) => {
```

File: webapp/src/viewModel/boardTree.ts [hunk 3]
```diff
@@ -55,57 +55,54 @@ class MutableBoardTree implements BoardTree {
...
+    usersById: {[key: string]: IUser}
 
     get allBlocks(): IBlock[] {
         return [this.board, ...this.views, ...this.allCards, ...this.cardTemplates, ...this.rawBlocks]
     }
 
-    constructor(board: MutableBoard) {
+    constructor(board: MutableBoard, usersById: {[key: string]: IUser}) {
         this.board = board
+        this.usersById = usersById
```
Comment (mgdelacroix) [inline webapp/src/viewModel/boardTree.ts:66]: Should this use `setUsersById(usersById)` instead of a direct assign?

**Review by mgdelacroix — APPROVED** [review]
Amazing work!! Feels very idiomatic and well structured :rocket: 

Just added a couple of minor nonblocking comments

File: webapp/src/viewModel/boardTree.ts (hunk 3 above)
Comment (jespino) [reply webapp/src/viewModel/boardTree.ts:66]: Not very important, actually that new function is not used anywhere, so I can remove the function.

File: webapp/src/store/currentWorkspace.ts (hunk 1 above)
Comment (jespino) [reply webapp/src/store/currentWorkspace.ts:25]: What do you mean?

**Pull request merged**